wordpress-mobile/WordPress-iOS
# Contact Support: Configure Chat Widget Design and Presentation logic

Fixes #21275

Configure chat widget design based on Android https://github.com/wordpress-mobile/WordPress-Android/pull/18930 and build upon that implementation to address some of the issues.

In the scope of this PR:
- Hide close button
- Hide the top header
- Hide floating button
- Make sure chatbot always appears
- Make sure chatbot history is always reset
- Set color to green to match Android

Other configuration will be done in the scope of other tasks:
- https://github.com/wordpress-mobile/WordPress-iOS/issues/21316
- https://github.com/wordpress-mobile/WordPress-iOS/issues/21315 


### Implementation

I took Android's implementation https://github.com/wordpress-mobile/WordPress-Android/pull/18930 and looked for alternative solutions to make sure the chatbot always appears after opening Contact Support.

The main issue as I found is us calling `DocsBot.open` and other docs-bot-related methods before `DocsBot` is properly loaded. It seems not enough to wait for `DocsBot.init` promise to complete, and even not enough to wait for `#docsbotai-root` to be mounted. We need to wait for `shadowRoot` element to be mounted and then we can perform actions on `DocsBot`. 

```mermaid
graph LR
    A[Observe #docsbotai-root` loaded] --> B[Observe ShadowRoot loaded]
    B --> F[Inject CSS that modifies docs bot UI]
    B -->H[Remove history]
    B --> E[Open DocsBot]
```

I have limited expertise in these solutions and relied on the solutions I found on the web. Based on my testing this solution looks to be performing better and not rely on timers to perform these actions. There's still danger in elements changing in ChatBot which would not allow us to hide header or close button but the main functionality of opening docs bot should work.


https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/348de7f4-f079-4f8a-a84b-8b6b44270591

### To test:

2. Open Jetpack
3. Enable Contact Support via DocsBot feature flag
4. Open Help&Support -> Contact Support
5. Confirm chat appears automatically
6. Confirm header, close button, and floating chat button are not visible
7. Interact with the chat, come back, and return
8. Confirm chat history is reset
9. Confirm chat color is green

## Regression Notes
1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:


UI Changes testing checklist:
